src/util/bbb.js

```diff
@@ -0,0 +1,35 @@
+const config = require('../../config.json').bbb;
+const crypto = require('crypto');
+const { XMLParser } = require('fast-xml-parser');
+
+if (!config.api_url.endsWith('/')) config.api_url += '/';
+
+const parser = new XMLParser();
+
+/**
+ * @param {'create' | 'join' | 'isMeetingRunning' | 'getMeetings' | 'getMeetingInfo'} endpoint
+ * @param {URLSearchParams | string} params
+ */
+const bbb = async (endpoint, params) => {
+    const search = new URLSearchParams(params);
+    const checksumString = endpoint + search.toString() + config.api_key;
+    const checksum = crypto.createHash('sha1').update(checksumString).digest('hex');
+
+    search.set('checksum', checksum);
+
+    const url = new URL(endpoint, config.api_url);
+    url.search = search;
+
+    if (endpoint === 'join') return url.toString();
```

Comment:
Instead of leaking the hash here, we could also set `redirect=false` and then redirect to the URL with the session token only.

Reply:
Hah, I wish! In my testing, setting `redirect` had no discernible effect at all. I tried quite a few potential alternatives to `false` as well, since I would also much prefer doing it that way, but no dice.

But I checked and Greenlight also "leaks" this hash, so *shrug*.